### PR TITLE
EN translation for passport tutorial

### PR DESCRIPTION
**Pull request by cemremengu**
Added passport tutorial English documentation  #363

### CODE REVIEW

File: docs/source/en/tutorials/passport.md [hunk 1]
```diff
@@ -1 +1,282 @@
-this document is still waiting for translation, see [Chinese Version](/zh-cn/tutorials/passport.html)
\ No newline at end of file
+## Title: Passport
+
+** "Login authentication" ** is a common business scenario, including "account password login method" and "third-party unified login."
```
Comment (atian25) [inline docs/source/en/tutorials/passport.md]: - 一个 `*` 且没空格
- `"` 要改为 ` 吧?

Comment (cemremengu) [reply docs/source/en/tutorials/passport.md]: 一个 * 且没空格 : do you mean  *`Login authentication`* ?

" 要改为 ` 吧?: I will replace all " with ` in this part

Comment (atian25) [reply docs/source/en/tutorials/passport.md]: `** "Login authentication" **` 的问题是 `** ` 后面多一个 * 和一个空格，这样是得不到粗体的效果的。如果直接用 ` 的话可以省掉 **

Comment (cemremengu) [reply docs/source/en/tutorials/passport.md]: Hopefully it's good now :)

**Comment by cemremengu** [comment]
@SunShinewyf 最后一个！

File: docs/source/en/tutorials/passport.md [hunk 2]
```diff
@@ -1 +1,282 @@
...
+
+### API
+
+[egg-passport](https://github.com/eggjs/egg-passport) provides the following extensions:
+
+* `ctx.user` - Get currently logged in user information
+* `ctx.isAuthenticated()` - Check if the request is authorized
+* `ctx.login(user, [options])` - Start a login session for the user
+* `ctx.logout()` - Exit and clear user information from session
+* `ctx.session.returnTo=` - Set redirect address after success before jumping to authentication
```
Comment (SunShinewyf) [inline docs/source/en/tutorials/passport.md]: 中文：“在跳转验证前设置，可以指定成功后的 redirect 地址”， 感觉不通顺：
set before jumping to authentication page, then can set redirect address after success

File: docs/source/en/tutorials/passport.md [hunk 3]
```diff
@@ -1 +1,282 @@
...
+
+The API also provides:
+
```
Comment (SunShinewyf) [inline docs/source/en/tutorials/passport.md]: also provides -> also be provided

File: docs/source/en/tutorials/passport.md [hunk 4]
```diff
@@ -1 +1,282 @@
...
+* `app.passport.deserializeUser(async (ctx, user) => {})` - Deserialize user information from the session
+* `app.passport.authenticate(strategy, options)` - Generate the specified authentication middleware
+    - `options.successRedirect` - specifies the redirect address after successful authentication
+    - `options.loginURL` - jump login address, defaults to `/passport/${strategy}`
+    - `options.callbackURL` - callback address after authorization, defaults to `/passport/${strategy}/callback`
+* `app.passport.mount(strategy, options)` - Syntactic sugar for developers to configure routing
+
+## Using Passport Ecosystem
+
+[Passport](http://www.passportjs.org/) has many middleware and it is impossible to carry out secondary packaging.
```
Comment (SunShinewyf) [inline docs/source/en/tutorials/passport.md]: impossible to carry out secondary packaging -> impossible to have the second encapsulation

File: docs/source/en/tutorials/passport.md [hunk 5]
```diff
@@ -1 +1,282 @@
...
+// app/router.js
+Module.exports = app => {
+  Const { router, controller } = app;
+  Router.get('/', controller.home.index);
+
+  // Callback page after successful authentication
+  Router.get('/authCallback', controller.home.authCallback);
+
+  // Render login page, user input account password
+  Router.get('/login', controller.home.login);
```
Comment (SunShinewyf) [inline docs/source/en/tutorials/passport.md]: user inputs

File: docs/source/en/tutorials/passport.md [hunk 6]
```diff
@@ -1 +1,282 @@
...
+  // Login verification
+  Router.post('/login', app.passport.authenticate('local', { successRedirect: '/authCallback' }));
+};
+```
+
+## How to develop an egg-passport plugin
+
+In the previous section, we learned how to use a Passport middleware in the framework. We can further package it as a plugin and give back to the community.
```
Comment (SunShinewyf) [inline docs/source/en/tutorials/passport.md]: further package -> further encapsulated

File: docs/source/en/tutorials/passport.md [hunk 7]
```diff
@@ -1 +1,282 @@
...
+
+**Configuration:**
+
+```js
+// {plugin_root}/config/config.default.js
+// https://github.com/jaredhanson/passport-local
+exports.passportLocal = {};
+```
+
+Note: [egg-passport](https://github.com/eggjs/egg-passport) standardizes the configuration fields, which are unified as `key` and `secret`, so if the corresponding Passport middleware attribute names are inconsistent, the developer should perform the conversion.
```
Comment (SunShinewyf) [inline docs/source/en/tutorials/passport.md]: perform -> do

File: docs/source/en/tutorials/passport.md [hunk 8]
```diff
@@ -1 +1,282 @@
...
+    // This does not process application-level logic and passes it to app.passport.verify for unified processing.
+    app.passport.doVerify(req, user, done);
+  }));
+};
+```
+
+[passport]: http://www.passportjs.org/
+[egg-passport]: https://github.com/eggjs/egg-passport
+[passport-local]: https://github.com/jaredhanson/passport-local
+[eggjs/examples/passport]: https://github.com/eggjs/examples/tree/master/passport
```
Comment (SunShinewyf) [inline docs/source/en/tutorials/passport.md]: 有一些代码片段首字母大写了~

File: docs/source/en/tutorials/passport.md [hunk 9]
```diff
@@ -1 +1,282 @@
...
+  // router.get('/passport/github', github);
+  // router.get('/passport/github/callback', github);
+}
+```
+
+### User Information Processing
+
+Then we also need:
+
+* When logging in for the first time, you generally need to put user information into the repository and record the Session.
```
[the remark this answers is not in the document]
Comment (cemremengu) [reply docs/source/en/tutorials/passport.md]: "logining in" is not really correct unfortunately. I changed it to "signing in" so it sounds better for you as well

File: docs/source/en/tutorials/passport.md [hunk 10]
```diff
@@ -1,20 +1,20 @@
 ## Title: Passport
 
-**`Login authentication`** is a common business scenario, including "account password login method" and "third-party unified login".
+**`Login authentication`** is a common business scenario, including "account password login" and "third-party unified login".
 
 Among them, we often use the latter, such as Google, GitHub, QQ unified login, which are based on [OAuth](https://oauth.net/2/) specification.
 
-[Passport](http://www.passportjs.org/) is a highly scalable authentication middleware that supports the `Strategy` of `Github` ,`Twitter`,`Facebook`, and other well-known service vendors. It also supports login and authorization verification via account passwords.
+[Passport](http://www.passportjs.org/) is a highly scalable authentication middleware that supports the `Strategy` of `Github` ,`Twitter`,`Facebook`, and other well-known service vendors. It also supports login authorization verification.
```
Comment (SunShinewyf) [inline docs/source/en/tutorials/passport.md]: via account passwords 这个还是要加上的~

File: docs/source/zh-cn/tutorials/passport.md [hunk 11]
```diff
@@ -248,7 +248,7 @@ $ egg-init --type=plugin egg-passport-local
 ```js
 // {plugin_root}/config/config.default.js
 // https://github.com/jaredhanson/passport-local
-exports.passportLocal = {
+module.exports.passportLocal = {
 };
 ```
```
Comment (SunShinewyf) [inline docs/source/zh-cn/tutorials/passport.md]: 该这一个地方就可以提交了~

**Comment by atian25** [comment]
这个继续改改？ 然后 rebase 下？

File: docs/source/zh-cn/tutorials/passport.md (hunk 11 above)
Comment (cemremengu) [reply docs/source/zh-cn/tutorials/passport.md]: Fixed

**Comment by cemremengu** [comment]
@SunShinewyf :eyes:

**Pull request merged**